travis.sh

```diff
@@ -64,18 +64,19 @@ if [ ! "$CATKIN_PARALLEL_TEST_JOBS" ]; then export CATKIN_PARALLEL_TEST_JOBS="$C
 if [ ! "$ROS_REPOSITORY_PATH" ]; then export ROS_REPOSITORY_PATH="http://packages.ros.org/ros-shadow-fixed/ubuntu"; fi
 if [ ! "$ROSDEP_ADDITIONAL_OPTIONS" ]; then export ROSDEP_ADDITIONAL_OPTIONS="-n -q -r --ignore-src"; fi
 if [ ! "$CATKIN_TOOLS_BUILD_OPTIONS" ]; then
-  if [ "ROS_DISTRO" = "hydro" ]; then
+  if [ "$ROS_DISTRO" = "hydro" ]; then
     export CATKIN_TOOLS_BUILD_OPTIONS="-iv --summarize --limit-status-rate 0.002"
   else
     export CATKIN_TOOLS_BUILD_OPTIONS="--summarize --limit-status-rate 0.002"
   fi
 fi
+if [ "$SUDO_PIP" = false ]; then export SUDO_PIP=""; else export SUDO_PIP="sudo -H"; fi
 echo "Testing branch $TRAVIS_BRANCH of $REPOSITORY_NAME"
 # Setup pip
 # FIXME: need to specify pip version to 6.0.7 to avoid unexpected error
 # https://github.com/jsk-ros-pkg/jsk_robot/pull/523#issuecomment-164699366
-sudo easy_install 'pip==6.0.7'
-sudo pip install -U -q pip setuptools
+$SUDO_PIP easy_install 'pip==6.0.7'
+$SUDO_PIP pip install -U -q pip setuptools
```

Comment:
> This is the specific behavior of rosdep. We didn't understand this behavior untill we executed rosdep install with duplicated arguments, right?

I'm not sure the user don't need to understand the behavior and it is only for developers in this case, because `--as-root pip:yes` is the default and unlikely specified by the user. Always the backward `--as-root` option is used and it is not difficult to understand...
Then could you please describe your tips of using `--as-root pip:no` for enabling the caching in README?
It's enough for me.

---

> Yes. This is what I mean. reusing sudo occurs unexpected effect. So it should not be overwritten in for example .travis.yml.

Ok, you don't like to reuse the `SUDO_PIP` environmental variable, but like to use only for `pip install` command. If so, why you need to overwirte `SUDO_PIP` as `sudo`? Why not below implementation?

```
if [ "$SUDO_PIP" != "false" ]; then
  sudo -H easy_install 'pip==6.0.7'
  pip install -U -q pip setuptools
else
  easy_install 'pip==6.0.7'
  pip install -U -q pip setuptools
fi
```

If you still like the current implementation, I think you should describe about the overwriting in the README.
Because the `BEFORE_SCRIPT` is used after the overwriting, and user need carefully use the `SUDO_PIP` environmental variable, as you pointed out about `ROSDEP_ADDITIONAL_OPTIONS` [here](https://github.com/jsk-ros-pkg/jsk_travis/pull/295#issuecomment-239088400).